mars/dataframe/expressions/arithmetic/add.py

```diff
@@ -48,7 +48,41 @@ def level(self):
     def fill_value(self):
         return self._fill_value
 
+    @classmethod
+    def constant_cls(cls):
+        return DataFrameAddConstant
+
+class DataFrameAddConstant(DataFrameOperand, DataFrameConstantMixin):
+    _op_type_ = OperandDef.ADD_CONSTANT
+
+    _axis = AnyField('axis')
+    _level = AnyField('level')
+    _fill_value = Float64Field('fill_value')
+
+    def __init__(self, axis=None, level=None, fill_value=None, **kw):
+        super(DataFrameAddConstant, self).__init__(_axis=axis, _level=level,
+                                                   _fill_value=fill_value, **kw)
+
+    @classproperty
+    def _operator(self):
+        return operator.add
+
+    @property
+    def axis(self):
+        return self._axis
+
+    @property
+    def level(self):
+        return self._level
+
+    @property
+    def fill_value(self):
+        return self._fill_value
 
 def add(df, other, axis='columns', level=None, fill_value=None):
     op = DataFrameAdd(axis=axis, level=level, fill_value=fill_value)
     return op(df, other)
+
+def add_constant(df, other, axis='columns', level=None):
```

Comment:
Just integrate the logic into `add`, if other is a constant value, generate `DataFrameAddConst` instead of `DataFrameAdd`.

Reply:
Tensor is doing the to constant part in https://github.com/mars-project/mars/blob/c87d4819e9712c70d1b6ff5e98f2e4b1cadfacaa/mars/tensor/expressions/arithmetic/core.py#L159-L162

Reply:
I would like also to explain a little bit that why we need a AddConstant op, that is because this op has only 1 input so that we can chain more op into one.